flake.nix

```diff
@@ -9,42 +9,37 @@
         "x86_64-linux" "i686-linux" "aarch64-linux"
         "x86_64-darwin" "aarch64-darwin"
       ];
-      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f system);
-
-      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; });
-
-      buildPackage = system: subPackages:
-        let
-          pkgs = nixpkgsFor.${system};
-          commit = builtins.substring 0 7 (self.rev or "dirty");
-          version = builtins.readFile ./VERSION;
-        in pkgs.buildGo120Module {
-          name = "go-waku";
-          src = self;
-          inherit subPackages;
-          tags = [ ];
-          ldflags = [
-            "-X github.com/waku-org/go-waku/waku/v2/node.GitCommit=${commit}"
-            "-X github.com/waku-org/go-waku/waku/v2/node.Version=${version}"
-          ];
-          doCheck = false;
-          # FIXME: This needs to be manually changed when updating modules.
-          vendorSha256 = "sha256-pNkh5ZJEhC/X/SoqfWcm3w7W1EGOT3m2dzePM/XuA9w=";
-          # Fix for 'nix run' trying to execute 'go-waku'.
-          meta = { mainProgram = "waku"; };
-        };
+      forEach = nixpkgs.lib.genAttrs;
+      forAllSystems = forEach supportedSystems;
+      pkgsFor = forEach supportedSystems (
+        system: import nixpkgs { inherit system; }
+      );
     in rec {
-      packages = forAllSystems (system: {
-        node    = buildPackage system ["cmd/waku"];
-        library = buildPackage system ["library/c"];
+      packages = forAllSystems (system: let
+        pkgs = pkgsFor.${system};
+        buildPackage = pkgs.callPackage ./default.nix;
+      in rec {
+        default = node;
+        node = buildPackage {
+          inherit self;
+          subPkgs = ["cmd/waku"];
+        };
+        static-library = buildPackage {
+          inherit self;
+          subPkgs = ["library/c"];
+          ldflags = ["-buildmode=c-archive"];
+          output = "libgowaku.a";
+        };
+        dynamic-library = buildPackage {
+          inherit self;
+          subPkgs = ["library/c"];
+          ldflags = ["-buildmode=c-shared"];
+          output = "libgowaku.a";
```

Comment:
Depending on the OS, it should be `.so` / `.dll` / `.dylib`
https://github.com/waku-org/go-waku/blob/master/Makefile#L19-L30

Reply:
Good point.

Reply:
Fixed.